File: src/main/java/starter/lec15/Lec15Main.java
package starter.lec15;

import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Lec15Main {

    public static void main(String[] args) {
        // Array
        int[] array = {100, 200};
        for (int i = 0; i < array.length; i++) {
            System.out.printf("%s %s\n", i, array[i]);
        }

        // List
        final List<Integer> numbers = Arrays.asList(100, 200);
        System.out.println(numbers.get(0));
        for (int number : numbers) {
            System.out.println(number);
        }
        for (int i = 0; i < numbers.size(); i++) {
            System.out.printf("%s %s", i, numbers.get(i));
        }

        // Map
        Map<Integer, String> oldMap = new HashMap<>();
        oldMap.put(1, "MONDAY");
        oldMap.put(2, "TUESDAY");

        Map.of(1, "MONDAY", 2, "TUESDAY");
    }
}

File: src/main/kotlin/starter/lec15/Lec15Main.kt
package starter.lec15

/**
 * 배열과 컬렉션을 다루는 방법
 */
fun main() {
    // Array
    val array = arrayOf(100, 200)

    for (i in array.indices) {
        println("$i ${array[i]}")
    }

    for ((idx, value) in array.withIndex()) {
        println("$idx $value")
    }

    // List
    val numbers = listOf(100, 200)
    val emptyList = emptyList<Int>()

    printNumbers(emptyList())

    println(numbers[0])

    for (number in numbers) {
        println(number)
    }

    for ((idx, value) in numbers.withIndex()) {
        println("$idx $value")
    }

    // Map
    val oldMap = mutableMapOf<Int, String>()
    oldMap[1] = "MONDAY"
    oldMap[2] = "TUESDAY"

    mapOf(1 to "MONDAY", 2 to "TUESDAY")

    for (key in oldMap.keys) {
        println(key)
        println(oldMap[key])
    }

    for ((key, value) in oldMap.entries) {
        println(key)
        println(value)
    }
}

// 타입을 추론할 수 있다면 제네릭 타입을 생략할 수 있다
private fun printNumbers(numbers: List<Int>) {

}
